appendices/memorymodel.txt

```diff
@@ -788,21 +788,53 @@ NOTE: It is possible for there to be a write between X and Y that overwrites
 a subset of the memory locations, but the remaining memory locations (M~2~)
 will still be visible-to Y.
 
-[[memory-model-scoped-modification-order-coherence]]
-== Scoped Modification Order Coherence
+[[memory-model-acyclicity]]
+== Acyclicity
+
+_Reads-from_ is a relation between operations, where the first operation is a
+write, the second operation is a read, and the second operation reads the
+value written by the first operation.
+_From-reads_ is a relation between operations, where the first operation is a
+read, the second operation is a write, and the first operation reads a value
+written earlier than the second operation in the second operation's scoped
+modification order (or the first operation reads from the initial value, and
+the second operation is any write to the same locations).
+
+Then the implementation must: guarantee that no cycles exist in the union of
+the following relations:
 
-Let A and B be mutually-ordered atomic operations, where A happens-before B,
-and let O be A's scoped modification order.
-Then:
+  * location-ordered
+  * scoped modification order (over all atomic writes)
+  * reads-from
+  * from-reads
 
-  * If A and B are both writes, then A must: be earlier than B in O
-  * If A and B are both reads, then the write that A takes its value from
-    must: be earlier in O than (or the same as) the write that B takes its
-    value from
+NOTE: This is a "consistency" axiom, which informally guarantees that
+sequences of operations can't violate causality.
```

Comment:
nit: should be a Note block.

Reply:
I'd like to keep this consistent with the rest of the appendix for now, but we can fix all of them in a later change if you'd like.